spec/models/miq_dialog/seeding_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'fileutils'
+
+describe MiqDialog do
+  describe "::Seeding" do
+    include_examples(".seed called multiple times", 20)
+
+    describe ".seed" do
+      let(:tmpdir)     { Pathname.new(Dir.mktmpdir) }
+      let(:dialog_dir) { tmpdir.join("product/dialogs/miq_dialogs") }
+      let(:dialog_yml) { dialog_dir.join("testing.yaml") }
+      let(:data_dir)   { Pathname.new(__dir__).join("data/product") }
+
+      before do
+        FileUtils.mkdir_p(dialog_dir)
+        FileUtils.cp_r(Rails.root.join("product/dialogs/miq_dialogs/miq_provision_dialogs.yaml"), dialog_dir, preserve: true)
+
+        stub_const("MiqDialog::Seeding::DIALOG_DIR", dialog_dir)
+      end
+
+      after do
+        FileUtils.rm_rf(tmpdir)
+      end
+
+      it "creates, updates, and changes records" do
+        described_class.seed
+
+        expect(MiqDialog.where(:name => "miq_provision_dialogs")).to exist
+
+        expect(MiqDialog.where(:name => "testing_dialog")).to_not exist
+
+        # Add new records
+        FileUtils.cp_r(data_dir, tmpdir, preserve: true)
+
+        described_class.seed
+
+        dialog = MiqDialog.find_by(:name => "testing_dialog")
+
+        expect(dialog).to have_attributes(
+          :name        => "testing_dialog",
+          :description => "Testing Dialog",
+          :filename    => "testing.yaml",
+          :file_mtime  => File.mtime(dialog_yml).utc.round,
+          :dialog_type => "MiqProvisionWorkflow",
+          :content     => a_hash_including(:dialog_order => %i[requester service])
+        )
+
+        # Update reports
+        dialog_mtime = dialog.file_mtime
+
+        # The mtime rounding is granular to the second, so need to be higher
+        # than that for test purposes
+        FileUtils.touch(dialog_yml, mtime: 1.second.from_now.to_time)
+
+        described_class.seed
+
+        expect(dialog.reload.file_mtime).to_not eq(dialog_mtime)
```

Comment:
I missed it on the reports PR but should we check that the existing `MiqDialog.where(:name => "miq_provision_dialogs")` mtime is unchanged after seeding here?

Reply:
Done.